### PR TITLE
Refactor DCLStatement to MySQL, Oracle, PostgreSQL, SQL92 and SQLServer DCLStatement

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #7170.

Changes proposed in this pull request:
- Refactor DCLStatement to MySQL, Oracle, PostgreSQL, SQL92 and SQLServer DCLStatement

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/test/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareCheckerTest.java [hunk 1]
```diff
@@ -81,12 +98,17 @@
...
             new CallStatement(), new ChangeMasterStatement(), new MySQLChecksumTableStatement(), new CommitStatement(), new CreateIndexStatement(), new DropIndexStatement(),
-            new CreateDatabaseStatement(""), new DropDatabaseStatement(""), new CreateTableStatement(mock(SimpleTableSegment.class)), new DropTableStatement(), new CreateUserStatement(),
-            new RenameUserStatement(), new DropUserStatement(), new CreateViewStatement(), new DropViewStatement(), new DeleteStatement(), new DoStatement(), new MySQLFlushStatement(),
-            new GrantStatement(), new InsertStatement(), new MySQLInstallPluginStatement(), new MySQLKillStatement(), new MySQLLoadIndexInfoStatement(), new MySQLOptimizeTableStatement(),
-            new RenameTableStatement(), new MySQLRepairTableStatement(), new MySQLResetStatement(), new RevokeStatement(), new SelectStatement(), new SetStatement(), new MySQLShowWarningsStatement(),
+            new CreateDatabaseStatement(""), new DropDatabaseStatement(""), new CreateTableStatement(mock(SimpleTableSegment.class)), new DropTableStatement(),
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/test/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareCheckerTest.java:104]: I suppose other `database` SQL  Statements except for `MySQL` are not wanted here. See the name of this class.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/test/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareCheckerTest.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/test/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/binary/prepare/MySQLComStmtPrepareCheckerTest.java:104]: > I suppose other `database` SQL Statements except for `MySQL` are not wanted here. See the name of this class.

@tristaZero I am very sorry for this mistake, and I will check this logic again. 🤣

**Review by tuohai666 — COMMENTED** [review]
Since this change is big, please perform the integration test using MySQL. 

1, env.properties:
databases=MySQL
mysql.host=127.0.0.1 (suppose your MySQL is on localhost)

2, go to shardingsphere-test-suite module

3, mvn clean install

**Comment by strongduanmu** [comment]
> Since this change is big, please perform the integration test using MySQL.
> 
> 1, env.properties:
> databases=MySQL
> mysql.host=127.0.0.1 (suppose your MySQL is on localhost)
> 
> 2, go to shardingsphere-test-suite module
> 
> 3, mvn clean install

@tuohai666 I have used the PR branch(`strongduanmu:issue-7170-dev-0904`) and the master branch to execute `mvn clean install` in the `shardingsphere-test-suite` module, and found that both branches have a lot of exceptions.
How can I test the impact of this PR? 🤔

**Comment by tuohai666** [comment]
> > Since this change is big, please perform the integration test using MySQL.
> > 1, env.properties:
> > databases=MySQL
> > mysql.host=127.0.0.1 (suppose your MySQL is on localhost)
> > 2, go to shardingsphere-test-suite module
> > 3, mvn clean install
> 
> @tuohai666 I have used the PR branch(`strongduanmu:issue-7170-dev-0904`) and the master branch to execute `mvn clean install` in the `shardingsphere-test-suite` module, and found that both branches have a lot of exceptions.
> How can I test the impact of this PR? 🤔

Hi, you can just run BaseDCLIT instead of mvn install.

**Comment by strongduanmu** [comment]
> > > Since this change is big, please perform the integration test using MySQL.
> > > 1, env.properties:
> > > databases=MySQL
> > > mysql.host=127.0.0.1 (suppose your MySQL is on localhost)
> > > 2, go to shardingsphere-test-suite module
> > > 3, mvn clean install
> > 
> > 
> > @tuohai666 I have used the PR branch(`strongduanmu:issue-7170-dev-0904`) and the master branch to execute `mvn clean install` in the `shardingsphere-test-suite` module, and found that both branches have a lot of exceptions.
> > How can I test the impact of this PR? 🤔
> 
> Hi, you can just run BaseDCLIT instead of mvn install.

@tuohai666 Thank you so much for your help. I have run `BaseDCLIT` on the pr branch and all the test cases have passed.

**Review by tristaZero — APPROVED** [review]

**Pull request merged**